Standardize sequence definitions

ibrahimWassouf opened the pull request:
## :bookmark_tabs: Summary

Migrates the Sequence Diagram to Typescript

Resolves #

## :straight_ruler: Design Decisions

Will follow the the guidelines and design choices made in #4501, #4486, and #4499 to maintain cohesiveness across the codebase as much as possible. I will attempt to refactor the code without changing functionality as much as possible. In a scenario where that may not be possible, I will maintain the old code and leave refactoring/optimizing to another issue/ticket.

### :clipboard: Tasks

Make sure you

- [ ] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [ ] :computer: have added necessary unit/e2e tests.
- [ ] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/community/development.md#3-update-documentation) is used for all new features.
- [ ] :bookmark: targeted `develop` branch

ibrahimWassouf committed:
Begin creating types for sequenceDB

ibrahimWassouf committed:
Add types to SequenceDB functions related to Boxes

ibrahimWassouf committed:
Add types to sequenceDB functions related to actors

Yokozuna59 commented on packages/mermaid/src/config.type.ts:
```diff
@@ -1087,7 +1087,7 @@ export interface SequenceDiagramConfig extends BaseDiagramConfig {
   diagramMarginX?: number;
   /**
-   * Margin to the over and under the sequence diagram
+   * Margin to the top and bottom of the sequence diagram
```
This is a generated file, please modify the original one: [`packages/mermaid/src/schemas/config.schema.yaml`](https://github.com/mermaid-js/mermaid/blob/6e51f8fd9840a788bcca1f2cd2aabd78bcf3b74a/packages/mermaid/src/schemas/config.schema.yaml#L1614)

ibrahimWassouf replied:
Oop rookie mistake

ibrahimWassouf committed:
Create new Message type and rename old interface to Text

ibrahimWassouf committed:
Add types to sequenceDB functions related to messages

ibrahimWassouf committed:
Change parseMessage to parseText

This allows for a conceptual and type splitting between texts which
appear in many places in the diagram (messages, notes,
actor descriptions, etc) and messages between participants.

I believe it makes more sense to call parseText rather than
parseMessage when used in context of working with other components
in the diagram which will help with developer experience.

A notable change when implementing this is changing all wrap
fields in relevant interfaces to optional. This is done so that
if left undefined, we will resolt to an autoWrap function which
references the diagram config. This will allow for an implementation
of a diagram wide default of either wrap or nowrap that's left to the
user to decide, instead of a hard-coded default.

ibrahimWassouf committed:
Add parseText to db interface

ibrahimWassouf committed:
Add types to db functions related to links

ibrahimWassouf committed:
Merge branch 'develop' into standardize-sequence-definitions

ibrahimWassouf committed:
Migrate enums as union types and add typing to addSignal function

ibrahimWassouf committed:
Add typed functions relating to wrap config

ibrahimWassouf committed:
Add types to addNote, addProperties, and addDetails

ibrahimWassouf committed:
Merge branch 'develop' into standardize-sequence-definitions

knsv commented:
Hi, and thanks for the effort you put into this PR. As part of some general housekeeping we're going through older draft PRs and closing ones that have been sitting idle for a while: this is purely to keep the queue manageable, not a judgment on the contribution. If you'd like to continue this work, you're very welcome to reopen the PR or open a new one based on current `develop`: we'll gladly take another look. Really appreciate you taking the time to contribute to Mermaid!
